applications/serial_lte_modem/doc/MQTT_AT_commands.rst

```diff
@@ -0,0 +1,296 @@
+
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XMQTTEVT: <evt_type>,<result>
+
+* The ``<evt_type>`` value is an integer.
+  When ``0``, it indicates the acknowledgment of the connection request.
+* The ``<result>`` value is an integer and it can assume the following values:
+
+  * ``0`` - Connection succeded.
+  * *Negative Value* - Error code. It indicates the reason for the failure.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XMQTTCON=1,"MyMQTT-Client-ID","","","mqtt.eclipse.org",1883
+   OK
+   #XMQTTEVT: 0,0
+
+::
+
+   AT#XMQTTCON=0
+   OK
+   #XMQTTEVT: 1,0
+
+
+Read command
+------------
+
+The read command shows MQTT client information.
+
+Syntax
+~~~~~~
+
+::
+
+   AT#XMQTTCON?
+
+
+
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XMQTTCON: <cid>,<username>,<password>,<url>,<port>[,<sec_tag>]]
+
+* The ``<url>`` value is a string and it indicates the MQTT broker hostname.
+* The ``<username>`` value is a string and it indicates the MQTT Client username.
+* The ``<password>`` value is a string and it indicates the MQTT Client password.
+* The ``<port>`` value is an integer and it indicates the MQTT broker port.
+* The ``<sec_tag>`` value is an integer and it indicates the credential of the security tag used for establishing a secure connection.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XMQTTCON?
+   #XMQTTCON: "MyMQTT-Client-ID","","","mqtt.eclipse.org",1883
+   OK
+
+Test command
+------------
+
+The test command is not supported.
+
+----
+
+MQTT Subscribe #XMQTTSUB
+========================
+
+The ``#XMQTTCON`` command allows you to subscribe to or unsubscribe from an MQTT topic.
+
+Set command
+-----------
+
+The set command allows you to subscribe to or unsubscribe from an MQTT topic.
+
+Syntax
+~~~~~~
+
+::
+
+   AT#XMQTTSUB=<topic>,<op>,<qos>
+
+
+* The ``<topic>`` parameter is a string and it indicates the topic to be subscribed to or unsubscribed from.
+* The ``<op>`` parameter is an integer and can accept one the following values:
+
+  * ``0`` - Subscribe to the topic.
+  * ``1`` - Unsubscribe from the topic.
+
+* The ``<qos>`` parameter is an integer and it indicates the MQTT Quality of Service types.
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XMQTTEVT: <evt_type>,<result>[,<topic>,<msg>]
+
+* The ``<evt_type>`` value is an integer and it can assume the following values:
+
+  * ``2`` - Notification that a *publish event* has been received on a topic the client is subscribed to.
+  * ``7`` - Acknowledgment of the subscribe request.
+  * ``8`` - Acknowledgment of the unsubscribe request.
+
+* The ``<result>`` value is an integer and it can assume the following values:
+
+  * ``0`` - Value indicating the acknowledgment of the connection request.
+  * *Negative Value* - Error code indicating the reason for the failure.
+
+* The ``<msg>`` value is a string and it contains the payload on the topic published.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XMQTTSUB="nrf91/slm/mqtt/topic0",1,0
```

Comment:
@lats1980 Is this syntax correct? Above we mentioned that the parameters should be in this order: <topic>,<op>,<qos>